src/coreclr/jit/codegencommon.cpp

```diff
@@ -2013,16 +2014,36 @@ void CodeGen::genEmitMachineCode()
     }
 
 #ifdef DEBUG
-    if (compiler->opts.disAsm || verbose)
+    const bool dspMetrics     = compiler->opts.dspMetrics;
+    const bool dspSummary     = compiler->opts.disAsm || verbose;
+    const bool dspMetricsOnly = dspMetrics && !dspSummary;
+
+    if (dspSummary || dspMetrics)
     {
-        printf("\n; Total bytes of code %d, prolog size %d, PerfScore %.2f, instruction count %d, allocated bytes for "
+        if (!dspMetricsOnly)
+        {
+            printf("\n");
+        }
+
+        printf("Total bytes of code %d, prolog size %d, PerfScore %.2f, instruction count %d, allocated bytes for "
```

Comment:
nit: PR changes this printf from `printf("\n; Total bytes..` to `printf("Total bytes..` and I believe the former is correct because it's JitDisasm comment so it has to starts with `;`

Reply:
Will fix in a subsequent change.

Reply:
jit-analyze is looking for this string (https://github.com/dotnet/jitutils/blob/e30e004fee30f6da62e2ddf856e31e677cec2955/src/jit-analyze/Program.cs#L294), so diffs are semi-broken now.

Reply:
Added that back in #97677